CHANGELOG.md

```diff
@@ -7,6 +7,7 @@
 ### Bug Fixes
 
 * When converting nullable BLOB field to required, `null` values should be converted to `byte[0]` instead of `byte[1]`.
+* `RealmObjectSchema.setRequired()` and `RealmObjectSchema.setNullable()` will throw when converting a nullable primary key field with null values stored to a required primary key field.
```

Comment:
Before it will convert `null` to `0` or `""` silently in the best case. the worst case it there was a `0` and `null`, after converting, the primary key contains two `0` values.